src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -577,6 +568,27 @@ class IntervalExpressionSegment(BaseSegment):
     ),
 )
 
+
+class SingleIdentifierSegment(BaseSegment):
+    """A SingleIdentifierSegment.
+
+    Override the default SingleIdentifierGrammar to allow two of these segments
+    separated by an at sign.
+
+    https://dev.mysql.com/doc/refman/8.0/en/account-names.html
+    https://dev.mysql.com/doc/refman/8.0/en/role-names.html
+    """
+
+    type = "single_identifier"
+    match_grammar: Matchable = ansi_dialect.get_grammar("SingleIdentifierGrammar").copy(
+        insert=[
+            Ref("SessionVariableNameSegment"),
+            Ref("SingleQuotedIdentifierSegment"),
+            Ref("DoubleQuotedLiteralSegment"),
+        ]
+    )
+
+
```

Comment:
Oh I see what you meant earlier now. There's a double `identifier`. No I don't think we should do that.

Unless you explicitly want the label, this should be a grammar, rather than a class with a `type`.

But I don't think it's worth having either here. So I'd delete this.

```suggestion
```